### ISSUE TITLE
Allow use of async return value promise

### ISSUE THREAD

**Issue by NilSet** [SonarSource/sonar-javascript#731]
**I want to report a bug.**

**SonarJS version:** 3.1.1<br>

**Rule key:** S3699 aka UseOfEmptyReturnValue

**Reproducer** 
```
var arrowReturnsPromise = async () => {
   var x = () => {return 1}
};
x = arrowReturnsPromise(); // False-positive
```

**Expected behavior**
All async functions return a promise, regardless of if the body has an empty return value, so this rule should allow use of the return value.
